File: app/src/main/java/ru/neosvet/ui/HelpAdapter.java
package ru.neosvet.ui;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import ru.neosvet.vestnewage.R;

public class HelpAdapter extends BaseAdapter {
    private Context context;
    private List<ListItem> data = new ArrayList<ListItem>();

    public HelpAdapter(Context context) {
        this.context = context;
    }

    public void clear() {
        data.clear();
    }

    public void addItem(String title, String des) {
        ListItem item = new ListItem(title);
        item.setDes(des);
        data.add(item);
    }

    @Override
    public int getCount() {
        return data.size();
    }

    @Override
    public ListItem getItem(int i) {
        return data.get(i); //data == null ? null :
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int pos, View convertView, ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(R.layout.item_detail, null);
        }
        TextView tv = (TextView) convertView.findViewById(R.id.text_item);
        tv.setText(data.get(pos).getTitle());
        tv = (TextView) convertView.findViewById(R.id.des_item);
        tv.setText(data.get(pos).getDes());
        if (data.get(pos).getCount() == 0)
            tv.setVisibility(View.GONE);
        else
            tv.setVisibility(View.VISIBLE);
        return convertView;
    }
}

File: app/src/main/java/ru/neosvet/vestnewage/view/list/HelpAdapter.kt
package ru.neosvet.vestnewage.view.list

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.view.isVisible
import androidx.recyclerview.widget.RecyclerView
import ru.neosvet.vestnewage.R
import ru.neosvet.vestnewage.data.HelpItem

class HelpAdapter(
    private val clicker: ((Int) -> Unit)
) : RecyclerView.Adapter<HelpAdapter.ViewHolder>() {

    companion object {
        private const val TYPE_ICON = 0
        private const val TYPE_DETAIL = 1
    }

    private val data = mutableListOf<HelpItem>()

    @SuppressLint("NotifyDataSetChanged")
    fun setItems(items: List<HelpItem>) {
        data.clear()
        data.addAll(items)
        notifyDataSetChanged()
    }

    fun update(index: Int, item: HelpItem) {
        data[index] = item
        notifyItemChanged(index)
    }

    override fun getItemCount(): Int = data.size

    override fun getItemViewType(position: Int): Int {
        return if (data[position].icon > 0)
            TYPE_ICON
        else
            TYPE_DETAIL
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return if (viewType == TYPE_ICON)
            ViewHolder(
                LayoutInflater.from(parent.context).inflate(R.layout.item_menu, null)
            )
        else ViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.item_detail, null)
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.setItem(data[position])
    }

    inner class ViewHolder(root: View) : RecyclerView.ViewHolder(root) {
        private val bgItem: View = root.findViewById(R.id.item_bg)
        private val tvTitle: TextView = root.findViewById(R.id.text_item)
        private val tvDes: TextView? = root.findViewById(R.id.des_item)
        private val ivIcon: ImageView? = root.findViewById(R.id.image_item)

        init {
            bgItem.setOnClickListener {
                clicker.invoke(layoutPosition)
            }
        }

        fun setItem(item: HelpItem) {
            tvTitle.text = item.title
            if (item.opened && item.content.isNotEmpty()) {
                tvDes?.isVisible = true
                tvDes?.text = item.content
                bgItem.setBackgroundResource(R.drawable.item_bg)
            } else {
                tvDes?.isVisible = false
                bgItem.setBackgroundResource(R.drawable.card_bg)
            }
            ivIcon?.setImageResource(item.icon)
        }
    }
}
